### PR TITLE
add blocknlpmodels in the docs

### PR DESCRIPTION
**Pull request by grishabh147**
Added details of `BlockNLPModels.jl` and `BlockNLPAlgorithms.jl` that implement `NLPModels.jl`'s API in the documentation.

### CODE REVIEW

File: docs/src/models.md [hunk 1]
```diff
@@ -24,6 +24,8 @@ If you want to create your own interface, check these [Guidelines](@ref).
   Creates a linear least squares model.
 - [PDENLPModels.jl](https://github.com/JuliaSmoothOptimizers/PDENLPModels.jl):
   For PDE-constrained problems.
+- [BlockNLPModels.jl](https://github.com/exanauts/BlockNLPModels.jl): For modeling block structured nonlinear optimization problems
+- [BlockNLPAlgorithms.jl](https://github.com/exanauts/BlockNLPAlgorithms.jl): Implements popular decomposition algorithms such as dual decomposition and ADMM to solve block strucutred NLPs.
```
Comment (tmigot) [inline docs/src/models.md:28]:
```suggestion
```

File: docs/src/models.md [hunk 2]
```diff
@@ -24,6 +24,8 @@ If you want to create your own interface, check these [Guidelines](@ref).
   Creates a linear least squares model.
 - [PDENLPModels.jl](https://github.com/JuliaSmoothOptimizers/PDENLPModels.jl):
   For PDE-constrained problems.
+- [BlockNLPModels.jl](https://github.com/exanauts/BlockNLPModels.jl): For modeling block structured nonlinear optimization problems
```
Comment (tmigot) [inline docs/src/models.md:27]:
```suggestion
- [BlockNLPModels.jl](https://github.com/exanauts/BlockNLPModels.jl): For modeling block structured nonlinear optimization problems.
```

**Review by tmigot — CHANGES_REQUESTED** [review]
Hey @grishabh147 !
Thank you for the PR, and very sorry to take such a long time before reviewing it.
I suggested removing the link to BlockNLPAlgorithms.jl as this is more a list of models than solver. But, I will definitely keep it in mind when creating a similar list for solvers.

Any plan to make the package a Julia registered package?

**Comment by grishabh147** [comment]
Hi @tmigot, I have made the requested changes, thanks for reviewing this pull-request.

Yes, there are plans to make this a registered package, however, we intend to do that once we have a better front-end for this. We are planning to integrate this with [Plasmo.jl](https://github.com/plasmo-dev/Plasmo.jl).

**Review by tmigot — APPROVED** [review]
LGTM, Thank you @grishabh147

**Pull request merged**